File: backend/IMS_API/models/SalesOrder.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
namespace IMS_API.Models;

public class SalesOrder
{
    public int Id { get; set; }
    [Required, MaxLength(30)]
    public string OrderNumber { get; set; } = null!;
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;

    public int CustomerId { get; set; }
    public Customer Customer { get; set; } = null!;
    public bool IsShipped { get; set; } = false;

    [Precision(18, 2)]
    public decimal Discount { get; set; } = 0;

    [Precision(18, 2)]
    public decimal Tax { get; set; } = 0;

    [Precision(18, 2)]
    public decimal TotalAmount { get; set; } = 0;

    public bool PaymentStatus { get; set; } = false;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    public ICollection<SalesOrderItem> Items { get; set; } = new List<SalesOrderItem>();
}

File: frontend/IMS_WEB/src/app/models/salesorders/salesorder.ts
export interface SalesOrder {
  id: number;
  orderNumber: string;

  customerId: number;
  customerName?: string;

  orderDate: Date;

  isShipped: boolean;
  discount: number;
  tax: number;
  totalAmount: number;

  createdAt?: Date;
  updatedAt?: Date | null;

  items?: SalesOrderItem[];
}

export interface SalesOrderCreate {
  salesOrder: {
    orderNumber: string;
    customerId: number;
    orderDate: Date;
    isShipped: boolean;
    discount: number;
    tax: number;
    totalAmount: number;
  };
  items: {
    productId: number;
    quantity: number;
    unitPrice: number;
    warehouseId: number;
    status: string;
  }[];
}

export interface SalesOrderWithItems {
  salesOrder: SalesOrder;
  items: SalesOrderItem[];
}

export interface SalesOrderItem {
  id: number;
  salesOrderId: number;

  productId: number;
  productName?: string;

  quantity: number;
  unitPrice: number;

  warehouseId: number;
  warehouseName?: string;

  status: string;

  createdAt: Date;
  updatedAt?: Date | null;
}
